Task2/classification.py

```diff
@@ -0,0 +1,203 @@
+import sys
+import numpy as np
+import pandas as pd
+from sklearn import preprocessing
+from sklearn import tree, linear_model, svm, naive_bayes
+from sklearn.metrics import precision_recall_fscore_support, average_precision_score, accuracy_score
+from sklearn.model_selection import GridSearchCV
+
+import warnings
+warnings.filterwarnings('ignore')  # "error", "ignore", "always", "default", "module" or "once"
+
+titles_final = ['Mr', 'Mrs', 'Ms', 'Master', 'Col', 'Sir', 'Lady', 'Dr', 'Rev']
+age_intervals = [6, 18, 32, 52, 72, 100]
+
+
+def age_to_group(age):
+    return float(next(x[0]+1 for x in enumerate(age_intervals) if x[1] >= age))
+
+
+def cross_validation_search(model, model_name, data, labels):
+    model.fit(data, labels)
+    if hasattr(model, 'best_params_'):
+        print("Best parameters set found on development set:")
+        print(model.best_params_)
+        print('Best score is: {}'.format(model.best_score_))
+    else:
+        model_pred = model.predict(data)
+        print('Best score is: {}'.format(accuracy_score(labels, model_pred)))
+
+
+def testing_models(model, model_name, data, labels, t_data, t_labels, final=False):
```

Comment:
Same as with cross_validation_search, why not creating a model_result object that contains the results of predictions